### PR TITLE
[Docs] add migration

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/en/migration.md [hunk 1]
```diff
@@ -2,16 +2,105 @@
...
+    - [Modular Design](#modular-design)
+      - [Datasets](#datasets-1)
+      - [Models](#models-1)
     - [Codebase Conventions](#codebase-conventions)
       - [Configs](#configs)
       - [Scripts](#scripts)
 
 ## Migration from MMSelfSup 0.x
 
+In this section, it explains some modifications of MMSelfSup 1.x.
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
In this section, we introduce some modifications of MMSelfSup 1.x., to help users to migrate their projects based on MMSelfSup from 0.x smoothly.
```

File: docs/en/migration.md [hunk 2]
```diff
@@ -2,16 +2,105 @@
...
+
+### Config
+
+This part shows the three main parts of our config files in `_base_` folder.
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
This section illusreates the changes of our config files in `_base_` folder, which includes three parts

- Datasets: `mmselfsup/configs/selfsup/_base_/datasets`
- Models: `mmselfsup/configs/selfsup/_base_/models`
- Schedules: `mmselfsup/configs/selfsup/_base_/schedules`

The detailed changes will be elobrated in the following.
```

File: docs/en/migration.md [hunk 3]
```diff
@@ -2,16 +2,105 @@
...
+            data_prefix='data/imagenet/train',
+            ann_file='data/imagenet/meta/train.txt',
+        ),
+        num_views=[1, 1],
+        pipelines=[train_pipeline1, train_pipeline2],
+        prefetch=prefetch,
+    ),
+    val=...)
+```
+
```
Comment (tonysy) [inline docs/en/migration.md:64]: data字段在1.x是否还存在，最好需要注明一下

File: docs/en/migration.md [hunk 4]
```diff
@@ -2,16 +2,105 @@
...
+Besides, we remove the key of `data_source` to keep the consistent pipeline format with other OpenMMLab projects. Please refer to [1_config.md](user_guides/1_config.md) for more details.
+
+#### Models
+
+In the config of models, there are two main different parts.
+
+One is that there is a new key called `data_preprocessor`, which is responsible for preprocessing the data, like normalization, channel conversion, etc.
+
```
Comment (tonysy) [inline docs/en/migration.md:112]: 可以加个Code snippet给个实例

File: docs/en/migration.md [hunk 5]
```diff
@@ -2,16 +2,105 @@
...
+The other one is that a new key `loss` in `head` to determine the loss function of the algorithm.
+
```
Comment (tonysy) [inline docs/en/migration.md:114]: 可以加个Code snippet给个实例， 在loss上注释一下是1.x引入的

File: docs/en/migration.md [hunk 6]
```diff
@@ -2,16 +2,105 @@
...
+#### Schedules
+
+| MMSelfSup 0.x    | MMSelfSup 1.x   | Remark                                                                                                                          |
+| ---------------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| optimizer_config | /               | It is removed.                                                                                                                  |
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
| optimizer_config | /               | Has been **removed**.                                                                                                                  |
```

File: docs/en/migration.md [hunk 7]
```diff
@@ -2,16 +2,105 @@
...
+| lr_config        | param_scheduler | The `param_scheduler` is a list to set learning rate or other parameters, which is more flexible.                               |
+| runner           | train_cfg       | The loop setting (`EpochBasedTrainLoop`, `IterBasedTrainLoop`) in `train_cfg` controls the work flow of the algorithm training. |
+
+### Folders and Files
+
+The table below records the general modification of the folders and files.
+
+| MMSelfSup 0.x         | MMSelfSup 1.x       | Remark                                                                                                                                                   |
+| --------------------- | ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| apis                  | /                   | Currently, the `apis` folder has been removed, it might be added in the future.                                                                          |
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
| apis                  | /                   | Currently, the `apis` folder has been **removed**, it might be added in the future.                                                                          |
```

File: docs/en/migration.md [hunk 8]
```diff
@@ -2,16 +2,105 @@
...
+| core                  | engine              | The `core` folder has been renamed to `engine`, which includes `hooks`, `opimizers`.                                                                     |
+| datasets              | datasets            | The datasets is implemented according to different datasets, such as ImageNet, Places205.                                                                |
+| datasets/data_sources | /                   | The `data_sources` has been removed and the directory of `datasets` now is consistent with other OpenMMLab projects.                                     |
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
| datasets/data_sources | /                   | The `data_sources` has been **removed** and the directory of `datasets` now is consistent with other OpenMMLab projects.                                     |
```

File: docs/en/migration.md [hunk 9]
```diff
@@ -2,16 +2,105 @@
...
+| datasets/pipelines    | datasets/transforms | The `pipelines` folder has been renamed to `transforms`.                                                                                                 |
+| /                     | evaluation          | The `evaluation` is created for some evaluation functions or classes, such as KNN function or layer for detection.                                       |
+| /                     | models/losses       | The `losses` folder is created to provide different loss implementation, which is from `heads`                                                           |
+| /                     | structures          | The `structures` folder is for the implementation of data structures. In MMSelfSup, we provide `selfsup_data_sample` to save different data information. |
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
| /                     | structures          | The `structures` folder is for the implementation of data structures. In MMSelfSup, we provide `selfsup_data_sample` to store different data information. |
```

File: docs/en/migration.md [hunk 10]
```diff
@@ -2,16 +2,105 @@
...
+| /                     | visualization       | The `visualization` folder contains the visualizer, which is responsible for some visualization tasks like visualizing data augment.                     |
```
Comment (tonysy) [inline docs/en/migration.md]:
```suggestion
| /                     | visualization       | The `visualization` folder contains the visualizer, which is responsible for some visualization tasks like visualizing data augmentation.                     |
```

**Review by tonysy — COMMENTED** [review]
As inline comments

File: docs/en/migration.md [hunk 11]
```diff
@@ -2,16 +2,136 @@
...
+    - [Modular Design](#modular-design)
+      - [Datasets](#datasets-1)
+      - [Models](#models-1)
     - [Codebase Conventions](#codebase-conventions)
       - [Configs](#configs)
       - [Scripts](#scripts)
 
 ## Migration from MMSelfSup 0.x
 
+we introduce some modifications of MMSelfSup 1.x, to help users to migrate their projects based on MMSelfSup from 0.x smoothly.
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
we introduce some modifications of MMSelfSup 1.x, to help users to migrate their projects based on MMSelfSup 0.x to 1.x smoothly.
```

File: docs/en/migration.md [hunk 12]
```diff
@@ -2,16 +2,136 @@
...
+
+### Config
+
+This section illusreates the changes of our config files in `_base_` folder, which includes three parts
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
This section illustrates the changes of our config files in `_base_` folder, which includes three parts
```

File: docs/en/migration.md [hunk 13]
```diff
@@ -2,16 +2,136 @@
...
+    dataset=dict(
+        type=dataset_type,
+        data_root=data_root,
+        ann_file='meta/train.txt',
+        data_prefix=dict(img_path='train/'),
+        pipeline=train_pipeline))
+val_dataloader = ...
+```
+
+Besides, we remove the key of `data_source` to keep the consistent pipeline format with other OpenMMLab projects. Please refer to [1_config.md](user_guides/1_config.md) for more details.
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
Besides, we remove the key of `data_source` to keep the pipeline format consistent with that in other OpenMMLab projects. Please refer to [1_config.md](user_guides/1_config.md) for more details.
```

File: docs/en/migration.md [hunk 14]
```diff
@@ -2,16 +2,136 @@
...
+| datasets              | datasets            | The datasets is implemented according to different datasets, such as ImageNet, Places205.                                                                 |
+| datasets/data_sources | /                   | The `data_sources` has been **removed** and the directory of `datasets` now is consistent with other OpenMMLab projects.                                  |
+| datasets/pipelines    | datasets/transforms | The `pipelines` folder has been renamed to `transforms`.                                                                                                  |
+| /                     | evaluation          | The `evaluation` is created for some evaluation functions or classes, such as KNN function or layer for detection.                                        |
+| /                     | models/losses       | The `losses` folder is created to provide different loss implementation, which is from `heads`                                                            |
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
| /                     | models/losses       | The `losses` folder is created to provide different loss implementations, which is from `heads`                                                            |
```

File: docs/en/migration.md [hunk 15]
```diff
@@ -2,16 +2,136 @@
...
+| /                     | structures          | The `structures` folder is for the implementation of data structures. In MMSelfSup, we provide `selfsup_data_sample` to store different data information. |
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
| /                     | structures          | The `structures` folder is for the implementation of data structures. In MMSelfSup, we implement a new data structure, `selfsup_data_sample`,  to pass and receive data throughout the training/val process. |
```

File: docs/en/migration.md [hunk 16]
```diff
@@ -2,72 +2,393 @@
...
-- The directory of `tools` is modified, thus it has more clear structure. It has several folders to manage different scripts. For example, it has two converter folders for models and data format. Besides, the benchmark related scripts are all in `benchmarks` folder, which has the same structure as `configs/benchmarks`.
-
-- The arguments in `train.py` has been updated. Two major modifications are listed below.
-
-  - Add `--cfg-options` to modify the config from cmd arguments.
-
-  - Remove `--pretrained` and use `--cfg-options` to set the pretrained models.
+We introduce some modifications of MMSelfSup 1.x, to help users to migrate their projects based on MMSelfSup 0.x to 1.x smoothly.
+
+MMSelfSup 1.x depends on some new packages, you can prepare a new clean environment and install again
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
MMSelfSup 1.x depends on some new packages, you should create a new environment, with these required packages installed.
```

File: docs/en/migration.md [hunk 17]
```diff
@@ -2,72 +2,393 @@
...
+according to the [install tutorial](./get_started.md). Or install the below packages manually.
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
according to the [install tutorial](./get_started.md). Three important packages are listed below,
```

File: docs/en/migration.md [hunk 18]
```diff
@@ -2,72 +2,393 @@
...
+
+1. [MMEngine](https://github.com/open-mmlab/mmengine): MMEngine is the core the OpenMMLab 2.0 architecture,
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
1. [MMEngine](https://github.com/open-mmlab/mmengine): MMEngine is the base of all OpenMMLab 2.0 repos,.
```

File: docs/en/migration.md [hunk 19]
```diff
@@ -2,72 +2,393 @@
...
+   and we splited many compentents unrelated to computer vision from MMCV to MMEngine.
```
Comment (YuanLiuuuuuu) [inline docs/en/migration.md]:
```suggestion
   and modules, which are not specific to Computer Vision, are migrated from MMCV to this repo.
```
